runtime/src/rent_collector.rs

```diff
@@ -35,8 +36,11 @@ impl RentCollector {
     // updates this account's lamports and status and returns
     //  the account rent collected, if any
     //
-    pub fn update(&self, account: &mut Account) -> u64 {
-        if account.executable || account.rent_epoch > self.epoch || sysvar::check_id(&account.owner)
+    pub fn update(&self, address: &Pubkey, account: &mut Account) -> u64 {
+        if account.executable
+            || account.rent_epoch > self.epoch
+            || sysvar::check_id(&account.owner)
+            || *address == incinerator::id()
```

Comment:
Nice! you actually addressed the rent thing?

... then, we need to gate this logic too. ;)

Otherwise, we're still vulnerable a dos while upgrading the cluster. That dos would only cost literally 1 lamports, giving it to `1nc1nerator11111111111111111111111111111111`, other than the normal 1-sig TX fee.